### PR TITLE
Fix dashboard filters get invisible while dragging & text selection while dragging

### LINKED ISSUE #12870: Text highlighting on Safari when reordering columns in Data Model tab
**Describe the bug**
When reordering columns on the Admin -> Data Model page, dragging columns inadvertently highlights text in other columns.

highlighted-text-on-column-reorder

Column reordering works as expected on Chrome, but this text highlighting issue occurs in Safari (and to some extent on Firefox).

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)

**To Reproduce**
See gif above. Must be in Safari. Firefox also has some highlighting issues.

**Expected behavior**
As on Chrome, dragging columns does not highlight text on the page.

**Severity**
P3 at most. Just an awkward, unexpected experience.

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1.1 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Mac OS X",
    "os.version": "10.15.5",
    "user.language": "en",
    "user.timezone": "America/Los_Angeles"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres",
      "bigquery"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "dev",
    "version": {
      "tag": "v0.36.0-snapshot",
      "hash": "8d3500b",
      "branch": "docs-data-model-and-localization",
      "date": "2020-06-23"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### LINKED ISSUE #18047: Dragging dashboards filters makes them hidden while dragging
**Describe the bug**
Dragging dashboards filters makes them hidden while dragging

**To Reproduce**
1. Go to dashboard - add a filter - try moving the filter, it becomes hidden while dragging and is then shown again on release

The filter gets `opacity:0; visibility:hidden;` while DND.

Similar happens in Native editor, but that also happens on 0.40.5

**Expected behavior**
Don't hide the filter while dragging. And plus points if z-index could be applied while dragging, so the widget being dragged is top.

**Information about your Metabase Installation:**
master `b557b69` - regression since 0.40.5, where it widget is shown while DND on dashboard

### PR DESCRIPTION
**Pull request by kulyk**
This PR fixes two issues related to sorting drag-n-drop UI behavior

1. Fixes #18047 — dashboard filters are hidden while dragging
2. Fixes text gets accidentally selected while dragging things around (#18032 followup). It fixes #12870

Both issues are related to components using [react-sortable-hoc](https://github.com/clauderic/react-sortable-hoc). This PR adds a thin wrapper for its `SortableContainer` HOC. The wrapper does two things:

* wraps the whole sortable list with a `div` with `user-select: none` style to prevent unwanted text selections (it's an open [issue](https://github.com/clauderic/react-sortable-hoc/issues/253) of `react-sortable-hoc)
* always passes a CSS class that would set `z-index: 1000;` to list's dragged element, so it always stays on top (fixes the hidden filters issue)

### To Verify

Follow the steps from demo videos below using **Firefox** and **Safari** (works fine with Chrome / Brave)

### Demo

**Before**

Here's a single example of an incorrect behavior with columns order. Other components have very similar behavior.

https://user-images.githubusercontent.com/17258145/134548143-8cd20a73-3692-40ec-aa55-4221a616e778.mov

**After**

<details>
<summary>Conditional Formatting</summary>

https://user-images.githubusercontent.com/17258145/134714111-a29a43b2-5a92-44c6-bb01-e46a1c963468.mov

</details>

<details>
<summary>Data Model</summary>

https://user-images.githubusercontent.com/17258145/134714123-ecc71ed3-aa86-4562-bc8b-dbae8dae7dd0.mov

</details>

<details>
<summary>Dashboard Filters</summary>

https://user-images.githubusercontent.com/17258145/134714194-60aee796-e334-4484-9bdd-0e9ed1433308.mov

</details>

### CODE REVIEW

File: frontend/src/metabase/components/sortable/sortable.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  );
+
+  // Makes sure the sortable-hoc always receives "react-sortable-hoc-helper" class
+  // The class just applies a big z-index to the dragged item created by sortable-hoc
+  // to prevent https://github.com/clauderic/react-sortable-hoc#item-disappearing-when-sorting--css-issues
+  function SortableListWithDraggedItemVisibilityFix({ helperClass, ...props }) {
+    return (
+      <SortableList
+        {...props}
+        helperClass={cx("react-sortable-hoc-helper", helperClass)}
```
Comment (gusaiani) [inline frontend/src/metabase/components/sortable/sortable.jsx:35]: Do you like this approach of drilling classNames enough to add more cases?

Versus going with just styled components for it?

Or is this a situation where you don't want to deeply change existing things in the codebase?

**Review by gusaiani — APPROVED** [review]
Yes, works better now 👍

A marginal comment inlined.

**Review by flamber — APPROVED** [review]
Nice!

File: frontend/src/metabase/components/sortable/sortable.jsx (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/components/sortable/sortable.jsx:35]: It's not a personal preference, but a part of `react-sortable-hoc's` API

**Pull request merged**